### DISCUSSION TITLE
using opc.https

### DISCUSSION THREAD

**Discussion by ZootHii** [eclipse-milo/milo#1025] [General]
Hello again, 

I want to use UA HTTPS as my connection address. I have tried these configurations 
```
public interface ClientExample2 {

    default String getEndpointUrl() {
//        return "opc.tcp://myip:53530/a";
        return "https://myip:53443";
    }

    default Predicate<EndpointDescription> endpointFilter() {
        return e -> getSecurityPolicy().getUri().equals(e.getSecurityPolicyUri());
    }

    default SecurityPolicy getSecurityPolicy() {
        return Arrays.stream(SecurityPolicy.values()).filter(sp -> sp.getUri().contains("None")).findFirst().get();
    }

    default MessageSecurityMode getSecurityMode() {
        return MessageSecurityMode.valueOf("None");
    }

    default IdentityProvider getIdentityProvider() {
        return new AnonymousProvider();
    }

    void run(OpcUaClient client, CompletableFuture<OpcUaClient> future) throws Exception;

}
```

but I got 

`status=Bad_ServiceUnsupported, message=The server does not support the requested service.`

Is there any example, can you provide me ? How can I connect to the server using https. By the way I don't have any problem with tcp.

Thanks a lot.

**Comment by kevinherron** [comment]
It looks like the server is returning that StatusCode for some reason, I'm not really sure why or what service it's in response to.

The HTTPS transport in Milo is not well tested and may not even be fully implemented. It also only supports the HTTPS + Binary encoding, which is not a common combination; most servers that support HTTPS use the XML encoding.

It's probably best if you just don't bother with the HTTPS transport. The TCP transport is the only one required by all OPC UA implementations and the only one you will find in use in the real world.
